test/integration_test.go

```diff
@@ -1198,6 +1199,67 @@ func TestAutoRetryEnvVarWithRecoverableError(t *testing.T) {
 	assert.NotContains(t, out.String(), "Apply complete!")
 }
 
+func TestAutomaticInitAfterRecoverableError(t *testing.T) {
+	rootPath := copyEnvironment(t, TEST_FIXTURE_AUTO_RETRY_APPLY_AUTOMATIC_INIT)
+
+	modulePath := util.JoinPath(rootPath, TEST_FIXTURE_AUTO_RETRY_APPLY_AUTOMATIC_INIT)
+
+	var paths = map[string]string{
+		"old":     util.JoinPath(modulePath, "module-old"),
+		"new":     util.JoinPath(modulePath, "module-new"),
+		"invalid": util.JoinPath(modulePath, "module-invalid"),
+	}
+
+	var copyTerragruntFiles = func(source string) error {
+		for _, file := range []string{"terragrunt.hcl", "main.tf"} {
+			if err := util.CopyFile(util.JoinPath(source, file), util.JoinPath(modulePath, file)); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	var (
+		err    error
+		stdOut bytes.Buffer
+		stdErr bytes.Buffer
+	)
+
+	assert.NoError(t, copyTerragruntFiles(paths["old"]))
```

Comment:
`assert.XXX` will mark the test as failed, but allow it to keep running. `require.XXX` will mark the test failed and stop the test at that point. In some of the checks here, such as this one, you probably want `require.XXX` instead of `assert.XXX`, as any subsequent output would just be confusing/misleading.